src/Npgsql/NpgsqlCommand.cs

```diff
@@ -846,14 +846,8 @@ void ValidateParameters(ConnectorTypeMapper typeMapper)
 
         #region Message Creation / Population
 
-        internal bool FlushOccurred { get; set; }
```

Comment:
This might be a little bit problematic - without synchronous flush on the first statement, we reduce the chance of getting an error. Although, it's not like it's different from async version...

Reply:
Always doing a flush right after the first statement would affect perf, so I don't think we should do it just to get better error behavior.

In any case, you're right that this is less useful without the flush, though there are a class of errors which happen before the flush. At the end of the day we need to do #1323 properly at some point.